corehq/apps/case_search/models.py

```diff
@@ -62,6 +63,28 @@ def get_or_none(self, **kwargs):
             return None
 
 
+class FuzzyProperties(models.Model):
+    domain = models.CharField(
+        max_length=256,
+        null=False,
+        blank=False,
+        db_index=True,
+    )
+    case_type = models.CharField(
+        max_length=256,
+        null=False,
+        blank=False,
+        db_index=True,
+    )
+    properties = ArrayField(
+        models.TextField(null=True, blank=True),
+        null=True,
+    )
+
+    class Meta(object):
+        unique_together = ('domain', 'case_type')
```

Comment:
probably not that necessary depending on the record but seems like index_together for `case_type` and `domain` would be the most efficient